### ISSUE TITLE
How to train on my own data

### ISSUE THREAD

**Issue by Iamyourbaba** [ellisdg/3DUnetCNN#249]
How to train on my own data. My data in the form of  Dicom, such as data in https://www.ircad.fr/research/3d-ircadb-01/

**Comment by ellisdg** [comment]
You can convert the DICOMS to nifti using [dcm2niix](https://github.com/rordenlab/dcm2niix/releases).

**Comment by sima97** [comment]
could some one help me?
could someone tell me the sequence of the code?

**Comment by DanpengDing** [comment]
> How to train on my own data. My data in the form of Dicom, such as data in https://www.ircad.fr/research/3d-ircadb-01/

hello，i also use this dataset for training. But i hava some problem：During the training, my loss was always 1

**Comment by ellisdg** [comment]
If the loss is always one, that could indicate that the training data is incorrect. Did you convert the images to Nifti? Have you examined the training images to see that they make sense?

**Comment by DanpengDing** [comment]
> If the loss is always one, that could indicate that the training data is incorrect. Did you convert the images to Nifti? Have you examined the training images to see that they make sense?

I converted CT data into nii
![image](https://github.com/ellisdg/3DUnetCNN/assets/93312605/1c98890a-d088-4f8d-a625-30dd42ab469d)

and I checked that my data and label corresponded correctly one-to-one.
![image](https://github.com/ellisdg/3DUnetCNN/assets/93312605/249f0c69-2a64-41b3-a4cb-fdd99ecca951)
![image](https://github.com/ellisdg/3DUnetCNN/assets/93312605/72ad5f6c-9810-476c-99e9-180461a104bf)

The above is the corresponding validation result.
The data is converted to nii format and the label is 255 instead of 1. I did not convert 255 to 1. Write "labels" directly to config.json: [
0,
255
].
Could this be the cause of the problem
this my config:
{
    "model": {
        "name": "DynUNet",
        "in_channels": 1,
        "out_channels": 2,
        "spatial_dims": 3,
        "deep_supervision": false,
        "strides": [
            [
                1,
                1,
                1
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ]
        ],
        "filters": [
            64,
            96,
            128,
            192,
            256,
            384
        ],
        "kernel_size": [
            [
                3,
                3,
                3
            ],
            [
                3,
                3,
                3
            ],
            [
                3,
                3,
                3
            ],
            [
                3,
                3,
                3
            ],
            [
                3,
                3,
                3
            ],
            [
                3,
                3,
                3
            ]
        ],
        "upsample_kernel_size": [
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ],
            [
                2,
                2,
                2
            ]
        ]
    },
    "optimizer": {
        "name": "Adam",
        "lr": 0.001
    },
    "loss": {
        "name": "DiceLoss",
        "include_background": false,
        "sigmoid": true
    },
    "cross_validation":
        {
            "n_folds": 5,
            "random_seed": 25
        },
    "scheduler": {
        "name": "ReduceLROnPlateau",
        "patience": 10,
        "factor": 0.5,
        "min_lr": 1e-08
    },
    "dataset": {
        "name": "SegmentationDatasetPersistent",
        "desired_shape": [
            128,
            128,
            128
        ],
        "labels": [
            0,
            255
        ],
        "setup_label_hierarchy": true,
        "normalization": "NormalizeIntensityD",
        "normalization_kwargs": {
            "channel_wise": true,
            "nonzero": false
        },
        "resample": true,
        "crop_foreground": true
    },
    "training": {
        "batch_size": 1,
        "validation_batch_size": 1,
        "amp": false,
        "early_stopping_patience": null,
        "n_epochs": 250,
        "save_every_n_epochs": null,
        "save_last_n_models": null,
        "save_best": true
    },
    "training_filenames": [
        {
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM1.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem1.nii"
        },    
 	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM4.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem4.nii"
        },  
	 {
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM5.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem5.nii"
        },  
 	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM6.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem6.nii"
        },  
 	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM7.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem7.nii"
        },  
	 {
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM8.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem8.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM9.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem9.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM10.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem10.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM11.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem11.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM16.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem16.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM17.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem17.nii"
        },  
	{
            "image": [
                "D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM19.nii"
            ],
            "label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem19.nii"
        }
    ]
}

**Comment by DanpengDing** [comment]
> If the loss is always one, that could indicate that the training data is incorrect. Did you convert the images to Nifti? Have you examined the training images to see that they make sense?

I converted CT data into nii
![image](https://github.com/ellisdg/3DUnetCNN/assets/93312605/64fb0c2d-f0b7-4415-b7dd-bda869019cf6)


and I checked that my data and label corresponded correctly one-to-one.

The data is converted to nii format and the label is 255 instead of 1. I did not convert 255 to 1. Write "labels" directly to config.json: [
0,
255
].
Could this be the cause of the problem
this my config:
{
"model": {
"name": "DynUNet",
"in_channels": 1,
"out_channels": 2,
"spatial_dims": 3,
"deep_supervision": false,
"strides": [
[
1,
1,
1
],
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
]
],
"filters": [
64,
96,
128,
192,
256,
384
],
"kernel_size": [
[
3,
3,
3
],
[
3,
3,
3
],
[
3,
3,
3
],
[
3,
3,
3
],
[
3,
3,
3
],
[
3,
3,
3
]
],
"upsample_kernel_size": [
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
],
[
2,
2,
2
]
]
},
"optimizer": {
"name": "Adam",
"lr": 0.001
},
"loss": {
"name": "DiceLoss",
"include_background": false,
"sigmoid": true
},
"cross_validation":
{
"n_folds": 5,
"random_seed": 25
},
"scheduler": {
"name": "ReduceLROnPlateau",
"patience": 10,
"factor": 0.5,
"min_lr": 1e-08
},
"dataset": {
"name": "SegmentationDatasetPersistent",
"desired_shape": [
128,
128,
128
],
"labels": [
0,
255
],
"setup_label_hierarchy": true,
"normalization": "NormalizeIntensityD",
"normalization_kwargs": {
"channel_wise": true,
"nonzero": false
},
"resample": true,
"crop_foreground": true
},
"training": {
"batch_size": 1,
"validation_batch_size": 1,
"amp": false,
"early_stopping_patience": null,
"n_epochs": 250,
"save_every_n_epochs": null,
"save_last_n_models": null,
"save_best": true
},
"training_filenames": [
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM1.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem1.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM4.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem4.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM5.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem5.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM6.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem6.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM7.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem7.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM8.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem8.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM9.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem9.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM10.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem10.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM11.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem11.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM16.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem16.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM17.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem17.nii"
},
{
"image": [
"D:/Net/3DUnetCNN/examples/nii/data/PATIENT_DICOM19.nii"
],
"label": "D:/Net/3DUnetCNN/examples/nii/label/venoussystem19.nii"
}
]
}
loss very high
![image](https://github.com/ellisdg/3DUnetCNN/assets/93312605/1e937cf0-da4b-4cb6-abb7-01da83fe7d34)
